sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFluxCore.java

```diff
@@ -0,0 +1,228 @@
+ * and individual items in such pages. This type supports user-provided continuation tokens,
+ * allowing for restarting from a previously-retrieved continuation token.
+ *
+ * The type is backed by the Page Retriever provider provided in it's constructor. The provider is
+ * expected to return {@link PageRetriever} when called. The provider is invoked for each Subscription
+ * to this Flux. Given provider is called per Subscription, the provider implementation can create
+ * one or more objects to store any state and Page Retriever can capture and use those objects.
+ * This indirectly associate the state objects to the Subscription. The Page Retriever can get called
+ * multiple times in serial fashion, each time after the completion of the Flux returned by the previous
+ * invocation. The final completion signal will be send to the Subscriber when the last Page emitted by
+ * the Flux returned by the Page Retriever has {@code null} continuation token.
+ *
+ * <p><strong>Extending PagedFluxCore for Custom Continuation Token support</strong></p>
+ * {@codesnippet com.azure.core.util.paging.pagedfluxcore.continuationtoken}
+ *
+ * @param <T> The type of elements in a {@link ContinuablePage}
+ * @param <P> The {@link ContinuablePage} holding items of type {@code T}.
+ *
+ * @see ContinuablePagedFlux
+ * @see ContinuablePage
+ */
+public abstract class ContinuablePagedFluxCore<C, T, P extends ContinuablePage<C, T>>
+    extends ContinuablePagedFlux<C, T, P> {
+    private final Supplier<PageRetriever<C, P>> pageRetrieverProvider;
+    private final int defaultPageSize;
+
+    /**
+     * Creates an instance of {@link ContinuablePagedFluxCore}.
+     *
+     * @param pageRetrieverProvider a provider that returns {@link PageRetriever}.
+     */
+    protected ContinuablePagedFluxCore(Supplier<PageRetriever<C, P>> pageRetrieverProvider) {
+        this.pageRetrieverProvider = Objects.requireNonNull(pageRetrieverProvider,
+            "'pageRetrieverProvider' function cannot be null.");
+        this.defaultPageSize = -1;
+    }
+
+    /**
+     * Creates an instance of {@link ContinuablePagedFluxCore}.
+     *
+     * @param pageRetrieverProvider a provider that returns {@link PageRetriever}.
+     * @param defaultPageSize the default preferred page size
+     * @throws IllegalArgumentException if defaultPageSize is not greater than zero
+     */
+    protected ContinuablePagedFluxCore(Supplier<PageRetriever<C, P>> pageRetrieverProvider, int defaultPageSize) {
+        this.pageRetrieverProvider = Objects.requireNonNull(pageRetrieverProvider,
+            "'pageRetrieverProvider' function cannot be null.");
+        if (defaultPageSize <= 0) {
+            throw new IllegalArgumentException("defaultPageSize > 0 required but provided: " + defaultPageSize);
+        }
+        this.defaultPageSize = defaultPageSize;
+    }
+
+    /**
+     * @return the page size configured for this {@link ContinuablePagedFluxCore}, -1 if unspecified.
```

Comment:
Can we instead use `Integer` instead to indicate unspecified by just setting it to `null`?

Reply:
agree, updated to use null instead of -1.